### ISSUE TITLE
`Error fetching metadata` when using `ghcr.io` for image subscription

### ISSUE THREAD

**Issue by morey-tech** [akuity/kargo#694]
Stage spec:
```
spec:
  promotionMechanisms:
    argoCDAppUpdates:
      - appName: morey-tech-demo-app-dev
        appNamespace: argocd
    gitRepoUpdates:
      - kustomize:
          images:
            - image: ghcr.io/morey-tech/demo-app
              path: env/dev
        readBranch: ''
        repoURL: 'https://github.com/morey-tech/demo-app-deploy'
        writeBranch: main
  subscriptions:
    repos:
      images:
        - repoURL: ghcr.io/morey-tech/demo-app
          updateStrategy: Latest
```

Error in status:
```
status:
  error: >-
    error syncing image repo subscriptions: error getting latest suitable tag
    for image "ghcr.io/morey-tech/demo-app": found no suitable version of image
    "ghcr.io/morey-tech/demo-app"
```

Error in controller logs:
```
time="2023-09-08T20:10:17Z" level=error msg="Error fetching metadata for morey-tech/demo-app:d5f4dd376524d10a7e0ca4b2b52c6426399c7749 - neither V1 or V2 or OCI manifest returned by registry: manifest unknown: OCI index found, but Accept header does not support OCI indexes"
```

Tested the same image pushed to Dockerhub and it works.
```
spec:
  subscriptions:
    repos:
      images:
      - repoURL: docker.io/moreytech/demo-app
        updateStrategy: Latest
status:
  availableFreight:
    - firstSeen: '2023-09-08T20:56:20Z'
      id: 315ebae132f84ff09d8e030aa4ed8ac6ef1cc145
      images:
        - repoURL: docker.io/moreytech/demo-app
          tag: 2bc20dfbfd6df6be278b9fd7bc187832839486ec
```

**Comment by morey-tech** [comment]
It seems like Kargo is re-using a portion from `argocd-image-updater`:
```
import (
	"github.com/argoproj-labs/argocd-image-updater/pkg/image"
	argoLog "github.com/argoproj-labs/argocd-image-updater/pkg/log"
	"github.com/argoproj-labs/argocd-image-updater/pkg/options"
	"github.com/argoproj-labs/argocd-image-updater/pkg/registry"
```
That project had the same issue: https://github.com/argoproj-labs/argocd-image-updater/issues/522 which issued a fix in `v0.12.2`.

[Kargo uses `v0.12.1`](https://github.com/akuity/kargo/blob/e6a23e16199f97bea60329f4a4d2625a4ab155ba/go.mod#L12), so bumping the minor version should fix this.

**Comment by krancour** [comment]
> which issued a fix in v0.12.2

@morey-tech thank you for doing the leg-work on this!

I'll try to bump us. iirc, we were pinned to an older version of image updater for some reason which I cannot recall and may no longer be relevant...

Going to at least give this a shot!
